source/common/upstream/subset_lb.cc

```diff
@@ -406,5 +414,32 @@ void SubsetLoadBalancer::HostSubsetImpl::update(const std::vector<HostSharedPtr>
                            filtered_added, filtered_removed);
 }
 
+SubsetLoadBalancer::PrioritySubsetImpl::PrioritySubsetImpl(const PrioritySet& original_priority_set)
+    : PrioritySetImpl(), original_priority_set_(original_priority_set) {
+  for (size_t i = 0; i < original_priority_set_.hostSetsPerPriority().size(); ++i) {
+    empty_ &= getOrCreateHostSet(i).hosts().empty();
+  }
+}
+
+HostSetPtr SubsetLoadBalancer::PrioritySubsetImpl::createHostSet(uint32_t priority) {
+  RELEASE_ASSERT(priority < original_priority_set_.hostSetsPerPriority().size());
```

Comment:
nit: This should probably be a normal ASSERT

Reply:
Happy to downgrade it in my follow-up but we'll crash on the next line regardless.  I just find crashes of "you have the wrong size" easier than "there's some invalid pointer dereference".

Reply:
It's not a big deal either way, but in the rest of the code this type of check would might likely be a normal debug assert.